sentinelhub/sentinelhub_byoc.py

```diff
@@ -93,17 +95,22 @@ def __init__(self, base_url=None, config=None):
 
     def iter_collections(self):
         """ Retrieve collections
-        `BYOC API reference <https://docs.sentinel-hub.com/api/latest/reference/#operation/getByocCollections>`
+
+        `BYOC API reference <https://docs.sentinel-hub.com/api/latest/reference/#operation/getByocCollections>`_
 
         :return: iterator over collections
         """
         url = f'{self.byoc_url}/collections'
-        return iter_pages(client=self.client, service_url=url,
-                          exception_message='Failed to obtain information about available collections.')
+        return SentinelHubFeatureIterator(
+            client=self.client,
+            url=url,
```

Comment:
Perhaps `url` can be constructed inline, e.g.:
```python
   url = f'{self.byoc_url}/collections'
```

(similarly below). Personal preference for short lines. Feel free to ignore.

Reply:
Yes, since it is a trivial operation we can have it directly in the constructor of `SentinelHubFeatureIterator`. Done